http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -119,12 +124,36 @@ private EndpointResponse handleTimeout(Throwable throwable) {
             endpointRequest.getPath())
         .timeout(httpActionOptions.getRequestTimeoutMs());
 
+    if (httpActionOptions.getResponseOptions().getPredicates().contains(JSON)) {
+      ResponsePredicate isJsonResponse = ResponsePredicate
+          .create(ResponsePredicate.JSON, result -> {
+            throw new ReplyException(ReplyFailure.RECIPIENT_FAILURE, result.message());
+          });
+
+      request.expect(io.vertx.reactivex.ext.web.client.predicate.ResponsePredicate
+          .newInstance(isJsonResponse));
+    }
+    attachResponsePredicatesToRequest(request,
+        httpActionOptions.getResponseOptions().getPredicates());
     endpointRequest.getHeaders().entries()
         .forEach(entry -> request.putHeader(entry.getKey(), entry.getValue()));
 
     return request.rxSend();
   }
 
+  private void attachResponsePredicatesToRequest(HttpRequest<Buffer> request,
+      Set<String> predicates) {
+    predicates.forEach(predicate -> {
```

Comment:
Your implementation is fine, for the future, you may also use more fluent approach with streams:

```suggestion
        predicates.stream()
        .filter(p -> !JSON.equals(p))
        .forEach(p -> {
          try {
            request.expect(predicatesProvider.fromName(p));
          } catch (NoSuchFieldException | IllegalAccessException e) {
            LOGGER.error("Cannot access ResponsePredicate identified by: {}", p);
          }
        });
```